### PR TITLE
[tools] Add concurrent read_segment_files validation option

### PR DESCRIPTION
**Pull request by robskillington**
**What this PR does / why we need it**:

Allows read_segment_files to be run in a pure validation mode and run concurrently (by default num CPUs / 2).

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/cmd/tools/read_index_segments/main/main.go [hunk 1]
```diff
@@ -87,34 +109,56 @@ func run(opts runOptions) {
...
+		out                io.Writer
+		validateWorkerPool xsync.WorkerPool
+	)
+	if !opts.validate {
+		// Only validating, output to dev null.
+		out = ioutil.Discard
+		validateWorkerPool = xsync.NewWorkerPool(opts.validateConcurrency)
+		validateWorkerPool.Init()
+		log.Info("validating segment files",
+			zap.Int("concurrency", opts.validateConcurrency))
```
Comment (arnikola) [inline src/cmd/tools/read_index_segments/main/main.go:122]: Do we not need the pool when writing to file?

File: src/cmd/tools/read_index_segments/main/main.go [hunk 2]
```diff
@@ -52,23 +64,33 @@ func main() {
 		golog.Fatalf("unable to create logger: %+v", err)
 	}
 
-	if *optPathPrefix == "" || *optOutputFile == "" {
+	if *optPathPrefix != "" && *optValidate {
```
Comment (arnikola) [inline src/cmd/tools/read_index_segments/main/main.go:67]: nit: maybe a little more readable if unwrap `*`s

File: src/cmd/tools/read_index_segments/main/main.go (hunk 1 above)
Comment (robskillington) [reply src/cmd/tools/read_index_segments/main/main.go:122]: Correct, no parallelism for writing to file (would cause issues).

**Review by arnikola — APPROVED** [review]
lgtm

**Pull request merged**